Code/Framework/AzFramework/AzFramework/Asset/AssetCatalog.cpp

```diff
@@ -808,96 +808,118 @@ namespace AzFramework
+                            {
+                                AzFramework::AssetCatalogEventBus::Broadcast(
+                                    &AzFramework::AssetCatalogEventBus::Events::OnCatalogAssetChanged, mapping);
+                            });
+                    }
                 }
-            }
-            else
-            {
-                AZ::SystemTickBus::QueueFunction([assetId]()
-                {
-                    AzFramework::AssetCatalogEventBus::Broadcast(&AzFramework::AssetCatalogEventBus::Events::OnCatalogAssetAdded, assetId);
-                });
-                for (const auto& mapping : message.m_legacyAssetIds)
+                else
                 {
-                    AZ::SystemTickBus::QueueFunction([mapping]()
+                    AZ::SystemTickBus::QueueFunction(
+                        [assetId]()
+                        {
+                            AzFramework::AssetCatalogEventBus::Broadcast(
+                                &AzFramework::AssetCatalogEventBus::Events::OnCatalogAssetAdded, assetId);
+                        });
+                    for (const auto& mapping : message.m_legacyAssetIds)
                     {
-                        AzFramework::AssetCatalogEventBus::Broadcast(&AzFramework::AssetCatalogEventBus::Events::OnCatalogAssetAdded, mapping);
-                    });
+                        AZ::SystemTickBus::QueueFunction(
+                            [mapping]()
+                            {
+                                AzFramework::AssetCatalogEventBus::Broadcast(
+                                    &AzFramework::AssetCatalogEventBus::Events::OnCatalogAssetAdded, mapping);
+                            });
+                    }
                 }
-            }
 
-            AzFramework::LegacyAssetEventBus::QueueEvent(AZ::Crc32(extension.c_str()), &AzFramework::LegacyAssetEventBus::Events::OnFileChanged, relativePath);
-            
-            if (AZ::Data::AssetManager::IsReady())
-            {
-                AZ::SystemTickBus::QueueFunction([assetId]()
+                AzFramework::LegacyAssetEventBus::QueueEvent(
+                    AZ::Crc32(extension.c_str()), &AzFramework::LegacyAssetEventBus::Events::OnFileChanged, relativePath);
+
+                if (AZ::Data::AssetManager::IsReady())
                 {
-                    AZ::Data::AssetManager::Instance().ReloadAsset(assetId, AZ::Data::AssetLoadBehavior::Default, true);
-                });
+                    AZ::SystemTickBus::QueueFunction(
+                        [assetId]()
+                        {
+                            AZ::Data::AssetManager::Instance().ReloadAsset(assetId, AZ::Data::AssetLoadBehavior::Default, true);
+                        });
+                }
+            }
+            else
+            {
+                AZ_TracePrintf("AssetCatalog", "AssetChanged: invalid asset id: %s\n", assetId.ToString<AZStd::string>().c_str());
```

Comment:
nit: Does not seem actionable.

Reply:
Updated a bit but these are kind of odd messages, they might as well be asserts because this is data coming from a network messages from AP, which is all coming from code.  I imagine if this error occurs its either some weird data corruption happened or most likely a coding error, so there's no user action to recommend.  Perhaps they should be upgraded to asserts?